spec/controllers/application_controller_spec.rb

```diff
@@ -1369,4 +1369,45 @@ def hub_path; end
       end
     end
   end
+
+  describe "#ensure_non_admins_reset_passwords_when_forced" do
+    let(:non_admin_user) { create(:team_member_user) }
+    let(:admin_user) { create(:admin_user) }
+
+    context "when a non-admin is logged in" do
+      before do
+        sign_in non_admin_user
+        non_admin_user.update!(forced_password_reset_at: nil)
+      end
+
+      it "redirects to change the password" do
+        get :index
+        expect(response).to redirect_to edit_hub_forced_password_resets_path
+      end
+
+      it "does not redirect if the password was already set" do
+        non_admin_user.update!(forced_password_reset_at: DateTime.now)
+        get :index
+        expect(response).not_to redirect_to edit_hub_forced_password_resets_path
+      end
+    end
+
+    context "when an admin is logged in" do
+      before do
+        sign_in admin_user
+      end
+
+      it "does not redirect to change password" do
+        get :index
+        expect(response).not_to redirect_to edit_hub_forced_password_resets_path
+      end
+    end
+
+    context "when no user is logged in" do
```

Comment:
Nice that you test the `nil` case! This test will let you find out if if you can safely remove the `&.` we suggested to remove in the implementation code.